Issue #3 - use 2d6 for a bell curve of die results rather linear

mhasko opened the pull request.

mhasko committed:
Issue #3 - use 2d6 for a bell curve of die results rather than a linear 1d10+2

gamefiend commented on moves/moves.go:
```diff
@@ -20,12 +20,17 @@ type Roll interface {
+	//roll the first d6, +1 since Intn is 0 based
 	rand.Seed(seed())
-	return ((rand.Intn(10)) + 2) + bonus
+	d1 := (rand.Intn(6) + 1)
```
I'm thinking: now that we have multiple rolls, what if the seed function becomes `die`, and returns `int` instead of int64?

Super quick concept:

```
func die(dieType int) int {
    // seed time code
    now := time.Now()
    rand.Seed(now.UnixNano())
    return rand(Intn(dieType) +1)
}
```

and then we can use the multiple dice more fluidly and more DRY in the actual Roll.

What do you think?

mhasko replied:
That's a lot better actually.  Don't know why I didn't do that the first time.  I'll blame new code environment.

gamefiend reviewed (changes requested):
Thanks for the PR! I like the change, but just want to change the nature of the die roll so that seeding happens every time a roll happens so that we don't need to remember to seed manually each time.

mhasko committed:
Issue 3: be smarter, create die func

create a Roll func that does all the dirty business of seeding hte prng, choosing a
prng based on the 'type' of die to be rolled, then use 2 of these to roll 2d6

gamefiend commented on moves/moves.go:
```diff
@@ -17,15 +17,17 @@ type Roll interface {
 func (m Move) Roll(bonus int) int {
-	rand.Seed(seed())
-	return ((rand.Intn(10)) + 2) + bonus
+	return (die(6) + die(6) + bonus)
```
awesome! thank you, gonna merge!

gamefiend reviewed (approved):
:ship:
